api/compaction_manager.cc

```diff
@@ -66,16 +66,18 @@ void set_compaction_manager(http_context& ctx, routes& r) {
     });
 
     cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<http::request> req) {
-        return ctx.db.map_reduce0([](replica::database& db) {
-            return do_with(std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>(), [&db](std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>& tasks) {
-                return do_for_each(db.get_column_families(), [&tasks](const std::pair<table_id, seastar::lw_shared_ptr<replica::table>>& i) -> future<> {
+        using task_map_type = std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>;
+        return ctx.db.map_reduce0([](replica::database& db) -> future<task_map_type> {
```

Comment:
I think you need coroutine::lambda here, since map_reduce0 isn't marked as lambda coroutine safe. It's not strictly necessary since the lambda doesn't capture anything, but better to be safe.